src/sentry/search/snuba/backend.py

```diff
@@ -165,13 +178,22 @@ def owner_filter(owner, projects):
                 is_active=True,
             ).values("team")
         )
-
-        owned_me = Q(
-            id__in=GroupOwner.objects.filter(
-                Q(user_id=owner[1].id) | Q(team__in=teams),
+        owned_by_me = GroupOwner.objects.filter(
+            Q(user_id=owner[1].id) | Q(team__in=teams),
+            project_id__in=[p.id for p in projects],
+            organization_id=organization_id,
+        )
```

Comment:
I think we can just add `group__groupassignee__isnull=True` here and skip the subquery. That will probably perform better than the in statement for projects with a lot of assignees, and we have some with 100k+ rows.

Reply:
Changed the query method to use this